docs/whitespace.md

```diff
@@ -0,0 +1,37 @@
+# Why we care about whitespace
```

Comment:
I see the motivation behind this whole file but I think this could be shortened to a paragraph or two and should really be a part of the style guide.